snap/snapcraft.yaml

```diff
@@ -569,6 +569,36 @@ parts:
       cmake -DCMAKE_INSTALL_PREFIX=$SNAPCRAFT_PART_INSTALL/ .
       make install
  
+  deck:
+    plugin: nil
+    build-packages:
+      - curl
+      - unzip
+    override-build: |
+      VERSION=1.8.2
+
+      # use dpkg architecture to figure out our target arch
+      case "$(dpkg --print-architecture)" in
+        amd64)
+          FILE_NAME=deck_${VERSION}_linux_amd64.tar.gz
+          # FILE_HASH=2be6414cdce1540c022acda76da55ef6bbd51c537dc2e3d4020652e72daec62d
+          ;;
+        arm64)
+          FILE_NAME=deck_${VERSION}_linux_arm64.tar.gz
+          # FILE_HASH=d4be8f68f0275701bcde71d130a997f5a32b47f91d5fb8a7eae7db666325a33b
+          ;;
+      esac
+      # download the archive and verify the checksum
+      # NOTE: the archive only includes the binary distribution,
+      # with no config or license files
```

Comment:
probably this means we need to manually add these ourselves

also did the project fully switch to using declarative config for Kong? I had not heard of deck before and just saw it here with your PR :-)

Reply:
No, the project hasn't (yet), but we can still use deck - it gets the configuration from the underlying Postgres database - and we can then import the configuration back in, which will populate the postgres 12 database. That saves us having to backup/restore the Postgres database itself - which is somewhat more complicated. I have a couple of more things to do before this is done - hence why it's still in draft :-)

Reply:
I ended up changing this The `deck` approach did end up having some issues so I went back to implementing this by dumping the postgres 10 database to a .sql file in the pre-refresh hook and then importing that back into a new postgres 12 database in the post-refresh hook.

Briefly, the main reason `deck` doesn't really work for us is:
- for `deck` to run and restore backed up data, Kong needs to be already running.
- I did try having a hook to run after the one shot security-proxy-setup service runs,  to add a call to “ $SNAP/bin/deck sync”
- That did in theory work, running after a new Kong installation has been set up on a new postgres 12 database, and restored the kong configuration
- However, the security-proxy-setup oneshot service has by that point created a new  kong configuration  with a new JWT token. When I run deck sync, the old configuration overrides that and the old JWT token is added to kong, whcih will cause issues.

So doing this with postgres directly in the pre-refresh and post-refresh hooks was a simpler and easier option.